integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonLoadDataCommand.scala

```diff
@@ -597,14 +602,18 @@ case class CarbonLoadDataCommand(
     }
     val partitionSchema =
       StructType(table.getPartitionInfo(table.getTableName).getColumnSchemaList.asScala.map(field =>
-      metastoreSchema.fields.find(_.name.equalsIgnoreCase(field.getColumnName))).map(_.get))
-
+        metastoreSchema.fields.find(_.name.equalsIgnoreCase(field.getColumnName))).map(_.get))
+    val overWriteLocal = if (overWrite && partition.nonEmpty) {
```

Comment:
In case of dynamic partition overwrite, existing partitions which are not present in current load won't be deleted so handle them.